services/exam.service.js

```diff
@@ -101,31 +126,38 @@ async function start(examId, userId) {
 	try {
 		const exam = await getById(examId);
 
-		// Sınav zamanı kontrolü
+		if (!exam) {
+			return { status: 404, message: "Exam not found" };
+		}
+
+		// Check exam times
 		const examTimeCheck = checkExamTimes(exam);
 		if (!examTimeCheck.valid) {
 			return { status: 400, message: examTimeCheck.message };
 		}
 
-		// Katılımcı kullanıcıyı kontrol et
-		return await participatedUserService.checkParticipation(
+		// Check or create participation
+		const response = await participatedUserService.checkParticipation(
 			userId,
 			examId,
-			{
-				createIfNotExist: true,
-			}
+			{ createIfNotExist: true }
 		);
+
+		return response;
 	} catch (error) {
 		console.log("Error starting exam: ", error);
-		return { status: 500, message: "Internal Server Error" };
+		throw new Error("Error starting exam");
 	}
 }
 
-// This function should move to question.service. It would be better.
-// These controls should be inside service or controller? Which one is more proper?
+// Q: This function should move to question.service. It would be better.
+// Q: These controls should be inside service or controller? Which one is more proper?
```

Comment:
Validation of the id is ok to do in the services, normally what we do as a pattern here is to use decorators, which actually a middleware that checks the id in the parameter, like you do in if(!exam) part and returns full exam or anything that is needed as DTO. But it is not that important here. Those are for big projects